README.md

```diff
@@ -1,46 +1,145 @@
 
-[![Build status](https://badge.buildkite.com/0f2d4956d49fbc795f9c17b0a741a6aa9ea532738e5f872ac8.svg?branch=master)](https://buildkite.com/shopify/kubernetes-deploy-gem)
+`kubernetes-deploy` is a command line tool that helps you ship changes to a Kubernetes namespace and understand the result. At Shopify, we use it within our much-beloved, open-source [Shipit](https://github.com/Shopify/shipit-engine#kubernetes) deployment app.
 
-Deploy script used to manage a Kubernetes application's namespace with [Shipit](https://github.com/Shopify/shipit-engine).
+This repo also includes related tools for [running tasks](#kubernetes-run) and [restarting deployments](#kubernetes-restart).
 
-## Installation
+![success](screenshots/success.png)
 
-Add this line to your application's Gemfile:
 
-```ruby
-gem 'kubernetes-deploy'
-```
 
-And then execute:
+![missing-secret-fail](screenshots/missing-secret-fail.png)
+
+
+
+
+## Highlights
+
+:eyes:	Watches the changes you requested to make sure they complete successfully.
+
+:interrobang:	Provides debug information for changes that failed.
+
+:1234:	Predeploys certain types of resources (e.g. ConfigMap, PersistentVolumeClaim) to make sure the latest version will be available when resources that might consume them (e.g. Deployment) are deployed.
+
+:closed_lock_with_key:	[Creates Kubernetes secrets from encrypted EJSON](#deploying-kubernetes-secrets-from-ejson), which you can safely commit to your repository
+
+:running:	[Runs tasks at the beginning of the deploy](#deploy-time-tasks) using pods (example use case: Rails migrations)
+
+
+
+--------
+
+
+
+## Table of contents
+
+**KUBERNETES-DEPLOY**
+* [Prerequisites](#prerequisites)
+* [Installation](#installation)
+* [Usage](#usage)
+  * [Using templates and variables](#using-templates-and-variables)
+  * [Running tasks at the beginning of a deploy](#running-tasks-at-the-beginning-of-a-deploy)
+  * [Deploying Kubernetes secrets (from EJSON)](#deploying-kubernetes-secrets-from-ejson)
+
+**KUBERNETES-RESTART**
+* [Usage](#usage-1)
+
+**KUBERNETES-RUN**
+* [Prerequisites](#prerequisites-1)
+* [Usage](#usage-2)
+
+**DEVELOPMENT**
+* [Setup](#setup)
+* [Running the test suite locally](#running-the-test-suite-locally)
+* [Releasing a new version (Shopify employees)](#releasing-a-new-version-shopify-employees)
+* [CI (Shopify employees)](#ci-shopify-employees)
+* [CI (External contributors)](#ci-external-contributors)
+
+**CONTRIBUTING**
+* [Contributing](#contributing)
+* [License](#license)
+
+
+
+----------
+
+
+
+## Prerequisites
+
+* Your cluster must be running Kubernetes v1.6.0 or higher
+* Each app must have a deploy directory containing its Kubernetes templates (see [Templates](#templates))
+* You must remove the` kubectl.kubernetes.io/last-applied-configuration` annotation from any resources in the namespace that are not included in your deploy directory. If you do not, kubernetes-deploy will delete them.
```

Comment:
That is the annotation that gets added when you create/update things via `kubectl apply`--I should probably mention that explicitly. It's what the pruning feature uses to identify which resources kubectl is authorized to manage. The easiest way to remove it is `kubectl edit` (and then remember  not to `kubectl apply` the thing in the future!).